api/api.go

```diff
@@ -428,50 +440,51 @@ func monetizedResourcesFilter(filterCondition string, billableTableName string,
 				b.org_guid,
 				b.space_guid,
 				b.memory_in_mb,
-				tstzrange(
-					greatest({{ .RangeFromPlaceholder }}, lower(vpp.valid_for), lower(b.duration)),
-					least({{ .RangeToPlaceholder }}, upper(vpp.valid_for), upper(b.duration))
-				) as duration,
+				r.request_range * vpp.valid_for * vcr.valid_for * b.duration as duration,
 				vpp.id AS pricing_plan_id,
 				vpp.name AS pricing_plan_name,
 				ppc.id AS pricing_plan_component_id,
 				ppc.name AS pricing_plan_component_name,
 				ppc.formula,
 				eval_formula(
 					b.memory_in_mb,
-					tstzrange(
-						greatest({{ .RangeFromPlaceholder }}, lower(vpp.valid_for), lower(b.duration)),
-						least({{ .RangeToPlaceholder }}, upper(vpp.valid_for), upper(b.duration))
-					),
+					r.request_range * vpp.valid_for * vcr.valid_for * b.duration,
 					ppc.formula
-				) as price_ex_vat,
+				) * vcr.rate as price_ex_vat,
 				eval_formula(
 					b.memory_in_mb,
-					tstzrange(
-						greatest({{ .RangeFromPlaceholder }}, lower(vpp.valid_for), lower(b.duration)),
-						least({{ .RangeToPlaceholder }}, upper(vpp.valid_for), upper(b.duration))
-					),
+					r.request_range * vpp.valid_for * vcr.valid_for * b.duration,
 					ppc.formula
-				) * (1 + vr.rate) as price_inc_vat,
+				) * vcr.rate * (1 + vr.rate) as price_inc_vat,
 				vr.name as vat_rate_name
 			from
 				authorized_resources b
+			cross join
+				request_range r
 			inner join
 				valid_pricing_plans vpp
-			on b.plan_guid = vpp.plan_guid
-				 and vpp.valid_for && b.duration
-				 and vpp.valid_for && tstzrange( {{ .RangeFromPlaceholder }}, {{ .RangeToPlaceholder }} )
+			on
+				b.plan_guid = vpp.plan_guid
+				and vpp.valid_for && b.duration
+			  and vpp.valid_for && r.request_range
+		  inner join
```

Comment:
I do not think the order is an issue in the final result. It would do a join with the currency rates for the given windows. But happy to change it to make it more readable.